### ISSUE TITLE
[Segment Replication] Review existing snapshot and restore compatibility with SegRep

### ISSUE THREAD

**Issue by dreamer-89** [opensearch-project/OpenSearch#3695]
As part of this issue, we want to identify if SegRep enabled cluster have any issues with snapshot and recovery. In case there are issues, identify changes needed to make it compatible. Concretely, we want to test snapshot/restore in below scenarios:
1. Snapshot & restore on segrep **disabled** cluster. 
2. Snapshot & restore on segrep **enabled** cluster. 
3. Snapshot on docrep cluster and restore on segrep cluster & vice-versa. 

Segment Replication
- Design Proposal #2229
- Meta Issue https://github.com/opensearch-project/OpenSearch/issues/2194

**Comment by dreamer-89** [comment]
Wrote segment replication integration tests for snapshot in commit below
https://github.com/dreamer-89/OpenSearch/commit/91aa30335d0bc6e07e2527546cc3cfff8d10e458

This is built on top of commit https://github.com/mch2/OpenSearch/commit/f99689403869de5a069358d8490752ed134d8cbf. Waiting for a working segrep changes before trying out a manual snapshot test.

**Comment by Rishikesh1159** [comment]
With opensearch 2.3 version, we have Segment replication feature as an experimental release. I have tested the latest 2.3 version with all the 3  above mentioned testing scenarios. The snapshot and restore process is compatible with segment replication in all above mentioned scenarios. Here is the process I used to test these:

Steps to Verify:
-> First I run a single node cluster.
-> Then I registered a repository (locally not on s3/remote store) for storing snapshots
-> Then I created an index of size 18gb with replicas and segment replication enabled.
-> Then I took a snapshot of the cluster.
-> Then I deleted the index.
-> Then I restore cluster from snapshot and i see deleted index came up again with segment replication enabled and entire 18gb size.
-> Later I tested in different scenarios by creating a segment replication enabled index and taking snapshot of cluster. Then I deleted the index and while restoring I restored the index with document replication enabled and entire process was successful. Similar result with vice-versa scenarios.
